sklearn/linear_model/_glm/glm.py

```diff
@@ -613,6 +613,10 @@ class GammaRegressor(_GeneralizedLinearRegressor):
     """Generalized Linear Model with a Gamma distribution.
 
     This regressor uses the 'log' link function.
+    Its optimization objective is::
+
+        sum(Xw_i + y_i * exp(-Xw_i) - log(y_i) - 1)
+        + (1/2) * alpha ||w||_2^2
```

Comment:
I am still not sure that we want to have the formulation here.

@lorentzenchr What is your feeling about it? If we start to introduce the objective function here, then we probably want to do the same for the other GLM.

Reply:
I personally like the objective as part of the docstring, but we often only have them in the user guide, examples: `LogisticRegression`, `PoissonRegressor`, `GammaRegressor`.

In case we include them in the docstring, having them as $\LaTeX$ would be much more readable. Therefore, having it in the Notes section would be numpydoc style compliant, https://numpydoc.readthedocs.io/en/latest/format.html#notes.

Reply:
Cons: It could be considered as implementation detail. It bloats the docstring. We would have it in 2 places, user guide and docstring.

Pros: Without objective, one does not know what the model does. Only the exact objective enables to compare with other libraries.